docs/src/main/asciidoc/pubsub.adoc

```diff
@@ -25,94 +25,130 @@ dependencies {
+===== Acknowledging messages
 
-To acknowledge multiple messages received from `pull()` or `pullAndConvert()` at once, you can use the `PubSubTemplate.ack()` method.
-You can also use the `PubSubTemplate.nack()` for negatively acknowledging messages.
+There are two ways to acknowledge messages.
 
+. To acknowledge multiple messages at once, you can use the `PubSubTemplate.ack()` method.
+You can also use the `PubSubTemplate.nack()` for negatively acknowledging messages.
 Using these methods for acknowledging messages in batches is more efficient than acknowledging messages individually, but they *require* the collection of messages to be from the same project.
 
-All `ack()`, `nack()`, and `modifyAckDeadline()` methods on messages as well as `PubSubSubscriberTemplate` are implemented asynchronously, returning a `ListenableFuture<Void>` to be able to process the asynchronous execution.
+. To acknowledge messages individually you can use the `ack()` or `nack()` method on each of them (to acknowledge or negatively acknowledge, correspondingly).
+
+NOTE: All `ack()`, `nack()`, and `modifyAckDeadline()` methods on messages as well as `PubSubSubscriberTemplate` are implemented asynchronously, returning a `ListenableFuture<Void>` to be able to process the asynchronous execution.
+
+==== JSON support
+
+For serialization and deserialization of POJOs using Jackson JSON, configure a `PubSubMessageConverter` bean, and the Spring Boot starter for GCP Pub/Sub will automatically wire it into the `PubSubTemplate`.
+
+[source,java,indent=0]
+----
+include::../../../src/test/java/org/springframework/cloud/gcp/autoconfigure/pubsub/it/PubSubTemplateDocumentationTests.java[tag=json_bean]
+----
+
+NOTE: Alternatively, you can set it directly by calling the `setMessageConverter()` method on the `PubSubTemplate`.
+Other implementations of the `PubSubMessageConverter` can also be configured in the same manner.
+
+Assuming you have the following class defined:
+
+[source,java,indent=0]
+----
+include::../../../src/test/java/org/springframework/cloud/gcp/autoconfigure/pubsub/it/PubSubTemplateDocumentationTests.java[tag=json_convertible_class]
+----
+
+You can serialize objects to JSON on publish automatically:
+
+[source,java,indent=0]
+----
+include::../../../src/test/java/org/springframework/cloud/gcp/autoconfigure/pubsub/it/PubSubTemplateDocumentationTests.java[tag=json_publish]
+----
+
+And thatt's how you convert messages to objects on pull:
+[source,java,indent=0]
+----
+include::../../../src/test/java/org/springframework/cloud/gcp/autoconfigure/pubsub/it/PubSubTemplateDocumentationTests.java[tag=json_pull]
+----
 
-`PubSubTemplate` uses a special subscriber generated by its `SubscriberFactory` to synchronously pull messages.
 
+Please refer to our link:../spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-json-sample[Pub/Sub JSON Payload Sample App] as a reference for using this functionality.
```

Comment:
This looks like a mistake from the previous version of the doc; I think this link is going to be broken if you test it with `mvn install -P=docs`. You will have to change to github link instead.